includes/RemovePIIJob.php

```diff
@@ -523,8 +523,15 @@ public function run() {
 		// Remove user email
 		$userLatest = $newName->getInstanceForUpdate();
 
-		if ( $userLatest->getEmail() ) {
-			$userLatest->invalidateEmail();
+		if ( $userLatest->getEmail() || $userLatest->getRealName() ) {
```

Comment:
Running the same command twice seems wasteful, can't this either be stored/cached or reduce to one call?

Reply:
Thanks! Done. Took a look and it seems saveSettings can even be called if no settings are changed, so I just left thar, removing the initial check, and always saveSettings.